homeassistant/components/netatmo/select.py

```diff
@@ -0,0 +1,161 @@
+        if home_id in hass.data[DOMAIN][DATA_SCHEDULES]
+    ]
+
+    _LOGGER.debug("Adding climate schedule select entities %s", entities)
+    async_add_entities(entities, True)
+
+
+class NetatmoScheduleSelect(NetatmoBase, SelectEntity):
+    """Representation a Netatmo thermostat."""
+
+    def __init__(
+        self, data_handler: NetatmoDataHandler, home_id: str, options: list
+    ) -> None:
+        """Initialize the sensor."""
+        SelectEntity.__init__(self)
+        super().__init__(data_handler)
+
+        self._home_id = home_id
+
+        self._data_classes.extend(
+            [
+                {
+                    "name": HOMEDATA_DATA_CLASS_NAME,
+                    SIGNAL_NAME: HOMEDATA_DATA_CLASS_NAME,
+                },
+            ]
+        )
+
+        self._device_name = self._data.homes[home_id]["name"]
+        self._attr_name = f"{MANUFACTURER} {self._device_name}"
+
+        self._model: str = "NATherm1"
+
+        self._attr_unique_id = f"{self._home_id}-schedule-select"
+
+        self._attr_current_option = (
+            self._data._get_selected_schedule(  # pylint: disable=protected-access
+                home_id=self._home_id
+            ).get("name")
+        )
+        self._attr_options = options
+
+    async def async_added_to_hass(self) -> None:
+        """Entity created."""
+        await super().async_added_to_hass()
+
+        for event_type in (EVENT_TYPE_SCHEDULE,):
+            self._listeners.append(
+                async_dispatcher_connect(
+                    self.hass,
+                    f"signal-{DOMAIN}-webhook-{event_type}",
+                    self.handle_event,
+                )
+            )
+
+    async def handle_event(self, event: dict) -> None:
+        """Handle webhook events."""
+        data = event["data"]
+
+        if self._home_id != data["home_id"]:
+            return
+
+        if data["event_type"] == EVENT_TYPE_SCHEDULE and "schedule_id" in data:
+            self._attr_current_option = self.hass.data[DOMAIN][DATA_SCHEDULES][
+                self._home_id
+            ].get(data["schedule_id"])
+            self.async_write_ha_state()
+
+    @property
+    def _data(self) -> pyatmo.AsyncHomeData:
+        """Return data for this entity."""
+        return cast(
+            pyatmo.AsyncHomeData,
+            self.data_handler.data[self._data_classes[0]["name"]],
+        )
+
+    async def async_select_option(self, option: str) -> None:
+        """Change the selected option."""
+        for sid, name in self.hass.data[DOMAIN][DATA_SCHEDULES][self._home_id].items():
+            if name == option:
+                await self._data.async_switch_home_schedule(
+                    home_id=self._home_id, schedule_id=sid
+                )
+                _LOGGER.debug(
+                    "Setting %s schedule to %s (%s)",
+                    self._home_id,
+                    option,
+                    sid,
+                )
```

Comment:
You want to reverse it so if it hangs, you know what was executed last. Also add a break to break out of the loop as it won't find the next option.

```suggestion
                _LOGGER.debug(
                    "Setting %s schedule to %s (%s)",
                    self._home_id,
                    option,
                    sid,
                )
                await self._data.async_switch_home_schedule(
                    home_id=self._home_id, schedule_id=sid
                )
                break
```

Bonus: you could rewrite it to a guard clause to save 1 indentation: `if name != option: continue`

Reply:
Good point, cheers.